### PR TITLE
Adding simplecov for Code Climate test coverage

### PR DESCRIPTION
**Pull request by abronkema**
# Checklist:

- I have performed a self-review of my own code,
- I have commented my code, particularly in hard-to-understand areas,
- I have made corresponding changes to the documentation,
- I have added tests that prove my fix is effective or that my feature works,
- New and existing unit tests pass locally with my changes ("bundle exec rake")

### Description
Adds [simplecov](https://github.com/colszowka/simplecov) gem to allow test coverage visibility via Code Climate (and via `/coverage/index.html`).

I added the `require 'simplecov'` to `spec_helper.rb`
   
List any dependencies that are required for this change. (gems, js libraries, etc.)

`simplecov` (added to Gemfile)

### Type of change

* New feature (non-breaking change which adds functionality)
* Documentation update

### How Has This Been Tested?

Once added to `spec_helper.rb`, ran Rspec to confirm tests worked. 

Result: 
image

### CODE REVIEW

**Comment by cattywampus** [comment]
Nice! Do we need to add any filters to the simplecov configuration in the `spec_helper.rb` to prevent things like the `spec/` folder or other areas from being calculated in the test coverage? I’m not aware of SimpleCov’s defaults, so it’s possible those are good enough... 🤷‍♂

**Comment by abronkema** [comment]
> Nice! Do we need to add any filters to the simplecov configuration in the `spec_helper.rb` to prevent things like the `spec/` folder or other areas from being calculated in the test coverage? I’m not aware of SimpleCov’s defaults, so it’s possible those are good enough... 🤷‍♂

@cattywampus good call! I've add `/spec` to the filter in `spec_helper.rb`, so it's now calculating coverage only for `app/` and `config/`. 

Thanks!

File: Gemfile [hunk 1]
```diff
@@ -3,6 +3,8 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 ruby '2.6.3'
 
+#add simpleCov for Code Climate test coverage visibility
+gem 'simplecov', require: false, group: :test
```
Comment (dixpac) [inline Gemfile]: We already have group test, starting on the line `65`...i think it would be good idea to move this also there :)

**Review by dixpac — COMMENTED** [review]
👍  this looks good to me. Down the road we can adjust the rules if necessary

File: Gemfile (hunk 1 above)
Comment (abronkema) [reply Gemfile]: Excellent - thanks for the callout! I've moved it to the `group :test` block and ran Rspec again to make sure it didn't break anything :) Thanks and let me know if anything else is needed.

File: spec/spec_helper.rb [hunk 2]
```diff
@@ -1,5 +1,8 @@
 # frozen_string_literal: true
-
+require 'simplecov'
+SimpleCov.start do
+  SimpleCov.add_filter '/spec'
```
Comment (cattywampus) [inline spec/spec_helper.rb:4]: Trying to think of other things that fall outside the realm of test coverage. Are things in the `bin/` and `db/` showing up in the report (I haven’t run it locally yet - on my phone)? I don’t think we would be writing tests to cover those.

Comment (abronkema) [reply spec/spec_helper.rb:4]: @cattywampus here's what I'm seeing when I run it locally:
image

**Review by cattywampus — APPROVED** [review]

**Pull request merged**